alpha/declcfg/declcfg.go

```diff
@@ -16,16 +16,18 @@ import (
 )
 
 const (
-	SchemaPackage = "olm.package"
-	SchemaChannel = "olm.channel"
-	SchemaBundle  = "olm.bundle"
+	SchemaPackage     = "olm.package"
+	SchemaChannel     = "olm.channel"
+	SchemaBundle      = "olm.bundle"
+	SchemaDeprecation = "olm.catalog.deprecation"
 )
 
 type DeclarativeConfig struct {
-	Packages []Package
-	Channels []Channel
-	Bundles  []Bundle
-	Others   []Meta
+	Packages     []Package
+	Channels     []Channel
+	Bundles      []Bundle
+	Deprecations map[string]Deprecation // mapping package name to deprecation
```

Comment:
Comment is superfluous if you name it `DeprecationsByPackage`?

Alternatively, why is this not a field on `Package`?

Reply:
It's not a field on Package because they are separate schema, and I wanted to avoid deprecations being written out under it when we encode it as json/yaml. 

I was arbitrarily not wanting a long variable name here, but you're right that it could be a self-documenting replacement for the comment.